src/sentry/incidents/logic.py

```diff
@@ -1103,6 +1103,18 @@ def delete_alert_rule(
 
         incidents = Incident.objects.filter(alert_rule=alert_rule)
         if incidents.exists():
+            # if this was a dynamic rule, delete the data in Seer
+            if alert_rule.detection_type == AlertRuleDetectionType.DYNAMIC:
```

Comment:
What if the rule never fired (meaning there were no incidents)? Can we put this outside of the if/else checks?

Reply:
If the rule never fired, then the Seer call will be in the post_delete()

Reply:
Ah is that what the `RegionScheduledDeletion` thing does, ok